optional-goi-analysis/goi-calculations.R

```diff
@@ -0,0 +1,291 @@
+## This script performs optional gene identifier mapping, as well as genes of 
+#  interest calculations in preparation for plotting.
+## Read in SingleCellExperiment RDS object that has been normalized and has PCA
+## embeddings.
+
+# Command line usage:
+
+# Rscript --vanilla goi-calculations.R \
+#   --sce "example-results/sample01/library01_miQC_processed_sce.rds" \
+#   --library_id "library01" \
+#   --seed 2021 \
+#   --input_goi_list "example-data/goi-lists/sample01_goi_list.tsv" \
+#   --perform_mapping TRUE \
+#   --sce_rownames_identifier "ENSEMBL" \
+#   --provided_identifier <- "SYMBOL" \
+#   --organism "Homo sapiens" \
+#   --multi_mappings "list" \
+#   --output_directory "example-results/sample01"
+
+## Set up -------------------------------------------------------------
+
+## Command line arguments/options
+
+# Library needed to declare command line options
+library(optparse)
+
+# Declare command line options
+option_list <- list(
+  optparse::make_option(
+    c("-i", "--sce"),
+    type = "character",
+    default = NULL,
+    help = "path to RDS file containing SCE object with log-normalized counts 
+    and PCA embeddings",
+  ),
+  optparse::make_option(
+    c("-l", "--library_id"),
+    type = "character",
+    default = NULL,
+    help = "Unique ID corresponding to library being analyzed",
+  ),
+  optparse::make_option(
+    c("-s", "--seed"),
+    type = "integer",
+    default = 2021,
+    help = "Random seed to set",
+    metavar = "integer"
+  ),
+  optparse::make_option(
+    c("-g", "--input_goi_list"),
+    type = "character",
+    default = NULL,
+    help = "file path to the input genes of interest list"
+  ),
+  optparse::make_option(
+    c( "--perform_mapping"),
+    action = "store_false",
+    help = "specifies whether or not to perform gene identifier mapping; the 
+    default is FALSE as mapping is not required when ENSEMBL ids are provided"
```

Comment:
```suggestion
    action = "store_true",
    default = FALSE,
    help = "specifies whether or not to perform gene identifier mapping; the 
    default is FALSE as mapping is not required when ENSEMBL ids are provided"
```
Based on the description I think what you mean here is to use `store_true` which will set this to true by using the `--perform_mapping` flag at the command line and then it will be false if the flag is not used.